tests/staticfiles_tests/tests.py

```diff
@@ -399,6 +399,71 @@ def test_ordering_override(self):
         self.assertFileContains('file2.txt', 'duplicate of file2.txt')
 
 
+# The collectstatic test suite already has conflicting files since both
+# project/test/file.txt and apps/test/static/test/file.txt are collected. To
+# properly test for the warning not happening unless we tell it to explicitly,
+# we only include static files from the default finders.
+@override_settings(
+    STATICFILES_DIRS=[],
+)
+class TestCollectionOverwriteWarning(CollectionTestCase):
+    """
+    Test warning in ``collectstatic`` output when a file is skipped because a
+    previous and separate file was already written to the same path. See
+    Command.collect in contrib/staticfiles/management/commands/collectstatic.py
+    """
+    def setUp(self):
+        # If this string is in the collectstatic output, it means the warning
+        # we're looking for was emitted.
+        self.warning_string = 'Found another file'
```

Comment:
this variable can be at the class level, no need to set it up for each test method.